### PR TITLE
[CMake] Adding C/C++ portion of CommandLine support

### PR DESCRIPTION
**Pull request by etcwilde**
Hooking up the bits of the build to link the swift commandline support C++ bits into swiftCore.

### CODE REVIEW

**Review by compnerd — APPROVED** [review]

**Comment by etcwilde** [comment]
@swift-ci please smoke test

**Comment by etcwilde** [comment]
@swift-ci please smoke test Linux

File: Runtimes/Core/core/CMakeLists.txt [hunk 1]
```diff
@@ -241,9 +241,10 @@ add_library(swiftCore
 
 set_target_properties(swiftCore PROPERTIES Swift_MODULE_NAME Swift)
 
-if(SwiftCore_ENABLE_COMMANDLINE)
+if(SwiftCore_ENABLE_COMMANDLINE_SUPPORT)
   # TODO: Pull in the CommandLine C++ files too
```
Comment (etcwilde) [inline Runtimes/Core/core/CMakeLists.txt:245]: remove this TODO

**Comment by etcwilde** [comment]
@swift-ci please smoke test

**Pull request merged**